walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/TimeRange.java

```diff
@@ -75,6 +75,27 @@ public int end() {
     return start + duration;
   }
 
+  /**
+   * Checks if one range starts on or before the other, and ends at the same time or after it.
+   */
+  public boolean startsDuringEndsOnOrAfter(TimeRange other) {
+    return (this.start > other.start) && (this.end() >= other.end());  
+  }
+
+  /**
+   * Checks if one range starts before the other, and ends during it.
+   */
+  public boolean startsOnOrBeforeEndsOnOrAfter(TimeRange other) {
+    return (this.start <= other.start) && (this.end() >= other.end());  
+  }
+
+  /**
+   * Checks if one range starts before the other, and ends during it.
```

Comment:
I think this function would return true for the following input of disjoint time ranges.      
```
|----|        |----|
 this          other
```
Don't you also need to enforce that the two intervals are overlapping?

Reply:
You're right, fixing this!